.github/workflows/build-install-test-snap-lxc.yaml

```
@@ -0,0 +1,83 @@
+name: build-install-test-snap-lxc
+
+on:
+  workflow_call:
+    inputs:
+      lxc-image:
+        required: true
+        type: string
+      ubuntu-image:
+        required: false
+        type: string
+        default: "ubuntu-latest"
+      snapcraft-channel:
+        required: false
+        type: string
+        default: "latest/stable"
+      snap-name:
+        required: true
+        type: string
+      branch-name:
+        required: true
+        type: string
+      snapcraft-args:
+        required: false
+        type: string
+      snap-install-args:
+        required: false
+        type: string
+      enable-experimental-extensions-env:
+        required: false
+        type: boolean
+      test-script:
+        description: "Custom script to run some custom snap testing"
+        required: false
+        type: string
+
+jobs:
+  build:
+    uses: ubuntu-robotics/snap_workflows/.github/workflows/build-install-test-snap.yaml@main
+    with:
+      ubuntu-image: ${{ inputs.ubuntu-image }}
+      snapcraft-channel: ${{ inputs.snapcraft-channel }}
+      branch-name: ${{ inputs.branch-name }}
+      snap-name: ${{ inputs.snap-name }}
+      snap-install-args: ${{ inputs.snap-install-args }}
+
+  test-on-lxc:
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+    - uses: actions/download-artifact@v3
+      name: ${{ inputs.snap-name }}-${{ inputs.branch-name }}
+      with:
+        path: .
+    - name: Setup LXD
+      uses: canonical/setup-lxd@v0.1.1
+      with:
+        channel: latest/stable
+    - name: Launch container
+      run: |
+        lxc launch '${{ inputs.lxc-image }}' lxc-container
+        sleep 10 ## TODO(investigate): without sleep the next steps are flaky
```

You probably should wait for `cloud-init` to finish initialising,

```suggestion
        lxc exec lxc-container -- cloud-init status --wait
```

---

Follow-up on not using `root`, you may want to wait for the `ubuntu` user. Something as the following may work,

```bash
until lxc exec lxc-container -- getent passwd ubuntu; do
  echo "Waiting for the ubuntu user."
  sleep 0.25
done
```